test/namedtuple.jl

```diff
@@ -76,6 +76,23 @@ let NT = NamedTuple{(:a,:b),Tuple{Int8,Int16}}, nt = (x=3,y=4)
     @test_throws MethodError convert(NT, nt)
 end
 
+@testset "convert NamedTuple" begin
+    conv1 = convert(NamedTuple{(:a,),Tuple{I}} where I, (;a=1))
+    @test conv1 == (a = 1,)
+
+    conv2 = convert(NamedTuple{(:a,),Tuple{Any}}, (;a=1))
+    @test conv2 == NamedTuple{(:a,), Tuple{Any}}((1,))
+
+    conv3 = convert(NamedTuple{(:a,),}, (;a=1))
+    @test conv3 == (a = 1,)
+
+    conv4 = convert(NamedTuple{(:a,),Tuple{I}} where I<:Unsigned, (;a=1))
+    @test conv4 == NamedTuple{(:a,), Tuple{Unsigned}}((0x0000000000000001,))
+
+    conv5 = convert(NamedTuple, (;a=1))
+    @test conv1 == (a = 1,)
```

Comment:
1. Use `===` to compare types as well as values
2. Use convert(Unsigned,1), since `0x00...` is sized for 64-bit machines
3. add a test_throws for convert(AbstractString, 1)
```suggestion
    @test conv1 === (a = 1,)

    conv2 = convert(NamedTuple{(:a,),Tuple{Any}}, (;a=1))
    @test conv2 === NamedTuple{(:a,), Tuple{Any}}((1,))

    conv3 = convert(NamedTuple{(:a,),}, (;a=1))
    @test conv3 === (a = 1,)

    conv4 = convert(NamedTuple{(:a,),Tuple{I}} where I<:Unsigned, (;a=1))
    @test conv4 === NamedTuple{(:a,), Tuple{Unsigned}}((1,))

    conv5 = convert(NamedTuple, (;a=1))
    @test conv1 === (a = 1,)
    
    res = @test_throws MethodError convert(NamedTuple{(:a,),Tuple{I}} where I<:AbstractString, (;a=1))
    @test res.value.f === :convert && res.value.args === (AbstractString, 1)
```

Reply:
Okay, will do. Also can you please explain me what `test_throws` does exactly? Especially here where you are testing the output of `test_throws` as well. Does it simply return the error associated with a function call?

Reply:
Hi, I think, in line 96 of your suggestion it should be:
```
@test conv_res.value.f === convert && conv_res.value.args === (AbstractString, 1)
```
instead of:
```
@test conv_res.value.f === :convert && conv_res.value.args === (AbstractString, 1)
```
I'm making the commit accordingly.